pandas/tests/frame/test_apply.py

```diff
@@ -1357,3 +1357,11 @@ def test_frequency_is_original(self, num_cols):
         df = DataFrame(1, index=index, columns=range(num_cols))
         df.apply(lambda x: x)
         assert index.freq == original.freq
+
+    def test_func_returns_object(self):
+        # GH 28652
+        df = DataFrame({"a": [1, 2]}, index=pd.Int64Index([1, 2]))
+        result = df.groupby("a").apply(lambda g: g.index)
+
+        tm.assert_index_equal(result[1], pd.Int64Index([1]))
+        tm.assert_index_equal(result[2], pd.Int64Index([2]))
```

Comment:
Hmmm...is there a reason why we can do something like `tm.assert_index_equal(result, expected)` ?

Reply:
Updated